src/main/java/ch/njol/skript/expressions/ExprItemWithPotionEffect.java

```diff
@@ -0,0 +1,177 @@
+				"[a] (4¦suspicious) stew with [potion [effect]] %potioneffecttypes% [potion [effect]] [[[of] tier] %-number%] [with duration %-timespan%]");
+		} else {
+			Skript.registerExpression(ExprItemWithPotionEffect.class, ItemType.class, ExpressionType.PROPERTY,
+				"[a] (0¦potion|1¦splash potion|2¦lingering potion|3¦tipped arrow) with [potion [effect]] %potioneffecttypes% [potion [effect]] [[[of] tier] %-number%] [with duration %-timespan%]",
+				"[a] (0¦potion|1¦splash potion|2¦lingering potion|3¦tipped arrow) with ambient [potion [effect]] %potioneffecttypes% [potion [effect]] [[[of] tier] %-number%] [with duration %-timespan%]",
+				"[a] (0¦potion|1¦splash potion|2¦lingering potion|3¦tipped arrow) with [potion [effect]] %potioneffecttypes% [potion [effect]] [[[of] tier] %-number%] without [any] particles [with duration %-timespan%]");
+		}
+	}
+	
+	@SuppressWarnings("null")
+	private Expression<PotionEffectType> potions;
+	@Nullable
+	private Expression<Number> tier;
+	@Nullable
+	private Expression<Timespan> duration;
+	private boolean ambient;
+	private boolean particles;
+	private int potion_type;
+	
+	
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		potions = (Expression<PotionEffectType>) exprs[0];
+		tier = (Expression<Number>) exprs[1];
+		duration = (Expression<Timespan>) exprs[2];
+		ambient = matchedPattern == 2;
+		particles = matchedPattern != 3;
+		potion_type = parseResult.mark;
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	protected ItemType[] get(Event e) {
+		ItemType item;
+		int a = 0;
+		if (tier != null) {
+			final Number amp = tier.getSingle(e);
+			if (amp != null)
+				a = amp.intValue() - 1;
+		}
+		int ticks = 15 * 20;
+		if (duration != null) {
+			final Timespan dur = duration.getSingle(e);
+			if (dur != null)
+				ticks = (int) (dur.getTicks_i() >= Integer.MAX_VALUE ? Integer.MAX_VALUE : dur.getTicks_i());
+		}
+		String name = null;
+		switch (potion_type) {
+			case 1:
+				item = Aliases.javaItemType("splash potion");
+				name = "Splash Potion";
+				break;
+			case 2:
+				item = Aliases.javaItemType("lingering potion");
+				name = "Lingering Potion";
+				break;
+			case 3:
+				item = Aliases.javaItemType("tipped arrow");
+				name = "Tipped Arrow";
+				break;
+			case 4:
+				item = Aliases.javaItemType("suspicious stew");
+				break;
+			default:
+				item = Aliases.javaItemType("potion");
+				name = "Potion";
+		}
+		ItemMeta meta = item.getItemMeta();
+		for (PotionEffectType potionEffectType : potions.getArray(e)) {
+			if (potion_type == 4) {
+				((SuspiciousStewMeta) meta).addCustomEffect(new PotionEffect(potionEffectType, ticks, 0), true);
+			} else {
+				((PotionMeta) meta).addCustomEffect(new PotionEffect(potionEffectType, ticks, a, ambient, particles), true);
+			}
+		}
+		if (name != null)
+			meta.setDisplayName(ChatColor.RESET + name);
```

Comment:
Why does the name have to be changed?

Reply:
I did this, because it defaulted to things like "Uncraftable Potion" and "Uncraftable Tipped Arrow". I figured this would look a little nicer for the defaults. The user can still rename them by using:
```vb
give player a potion with slowness named "Slow Potion"
```

Reply:
that won't use the locale name and might cause hassle/bugs